lib/block_view_validator_test.go

```diff
@@ -924,65 +933,127 @@ func TestGetTopValidatorsByStakeMergingDbAndUtxoView(t *testing.T) {
 	require.NoError(t, utxoView.FlushToDb(blockHeight))
 
-	// Fetch m1 so it is also cached in the UtxoView.
-	validatorEntry, err = utxoView.GetValidatorByPKID(m1PKID)
+	// Verify m1 is stored in the db.
+	validatorEntry, err = DBGetValidatorByPKID(db, chain.snapshot, m1PKID)
 	require.NoError(t, err)
 	require.NotNil(t, validatorEntry)
+	require.Equal(t, validatorEntry.TotalStakeAmountNanos, uint256.NewInt().SetUint64(400))
+	require.Equal(t, validatorEntry.Status(), ValidatorStatusJailed)
 
-	// Verify m1 is stored in the db.
-	validatorEntry, err = DBGetValidatorByPKID(db, chain.snapshot, m1PKID)
+	// Store m2's ValidatorEntry in the db with TotalStake = 300 nanos.
+	m2ValidatorEntry := &ValidatorEntry{
+		ValidatorPKID:         m2PKID,
+		TotalStakeAmountNanos: uint256.NewInt().SetUint64(300),
+	}
+	utxoView._setValidatorEntryMappings(m2ValidatorEntry)
+	require.NoError(t, utxoView.FlushToDb(blockHeight))
+
+	// Verify m2 is stored in the db.
+	validatorEntry, err = DBGetValidatorByPKID(db, chain.snapshot, m2PKID)
 	require.NoError(t, err)
 	require.NotNil(t, validatorEntry)
-	require.Equal(t, validatorEntry.TotalStakeAmountNanos, uint256.NewInt().SetUint64(200))
+	require.Equal(t, validatorEntry.TotalStakeAmountNanos, uint256.NewInt().SetUint64(300))
 
-	// Verify m1 is also stored in the UtxoView.
+	// Store m3's ValidatorEntry in the db with TotalStake = 600 nanos.
+	m3ValidatorEntry := &ValidatorEntry{
+		ValidatorPKID:         m3PKID,
+		TotalStakeAmountNanos: uint256.NewInt().SetUint64(600),
+	}
+	utxoView._setValidatorEntryMappings(m3ValidatorEntry)
+	require.NoError(t, utxoView.FlushToDb(blockHeight))
+
+	// Verify m3 is stored in the db.
+	validatorEntry, err = DBGetValidatorByPKID(db, chain.snapshot, m3PKID)
+	require.NoError(t, err)
+	require.NotNil(t, validatorEntry)
+	require.Equal(t, validatorEntry.TotalStakeAmountNanos, uint256.NewInt().SetUint64(600))
+
+	// Fetch m2 so it is also cached in the UtxoView.
+	validatorEntry, err = utxoView.GetValidatorByPKID(m2PKID)
+	require.NoError(t, err)
+	require.NotNil(t, validatorEntry)
+
+	// Verify m2 is also stored in the UtxoView.
 	require.Len(t, utxoView.ValidatorMapKeyToValidatorEntry, 1)
-	require.Equal(t, utxoView.ValidatorMapKeyToValidatorEntry[validatorEntry.ToMapKey()].ValidatorPKID, m1PKID)
+	require.Equal(t, utxoView.ValidatorMapKeyToValidatorEntry[m2ValidatorEntry.ToMapKey()].ValidatorPKID, m2PKID)
 	require.Equal(
 		t,
-		utxoView.ValidatorMapKeyToValidatorEntry[validatorEntry.ToMapKey()].TotalStakeAmountNanos,
-		uint256.NewInt().SetUint64(200),
+		utxoView.ValidatorMapKeyToValidatorEntry[m2ValidatorEntry.ToMapKey()].TotalStakeAmountNanos,
+		uint256.NewInt().SetUint64(300),
 	)
 
-	// Store m2's ValidatorEntry in the UtxoView.
-	m2ValidatorEntry := &ValidatorEntry{
-		ValidatorPKID:         m2PKID,
+	// Store m3's ValidatorEntry in the UtxoView with isDeleted=true.
+	m3ValidatorEntry.isDeleted = true
+	utxoView._setValidatorEntryMappings(m3ValidatorEntry)
```

Comment:
should we call `_deleteValidatorEntryMappings` instead?
```suggestion
	utxoView._deleteValidatorEntryMappings(m3ValidatorEntry)
```